doc/reference/optimization.rst

```diff
@@ -0,0 +1,91 @@
+Optimization techniques for Plutus scripts
+==========================================
+
+Identifying problem areas
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In order to identify which parts of the script are responsible for significant resource consumption, you can use the :ref:`profiling support <profiling_scripts>`.
+
+Using strict let-bindings to avoid recomputation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Comment:
Perhaps be even more clear that unlike Haskell's let that allows sharing (call-by-need) , our non-strict let do not allow sharing (call-by-name). Hence the recomputations.